### PR TITLE
Fix client analytics

### PR DESCRIPTION
**Pull request by atavism**
Resolves https://github.com/getlantern/lantern/issues/2788

### CODE REVIEW

**Comment by myleshorton** [comment]
This bizarrely still has the issue of no cid for me:

```
Jul 24 04:32:49.756 - 0m0s DEBUG analytics: analytics.go:150 Full analytics request: POST /collect HTTP/1.1
Jul 24 04:32:49.756 - 0m0s Host: ssl.google-analytics.com
Jul 24 04:32:49.756 - 0m0s User-Agent: Go 1.1 package http
Jul 24 04:32:49.756 - 0m0s Content-Length: 60
Jul 24 04:32:49.756 - 0m0s Content-Type: application/x-www-form-urlencoded
Jul 24 04:32:49.756 - 0m0s Accept-Encoding: gzip
Jul 24 04:32:49.756 - 0m0s
Jul 24 04:32:49.756 - 0m0s _v=j12&sr=1440x900&t=pageview&tid=UA-21815217-2&ul=en-us&v=1
```

I'm just adding this in the SendRequest method (oh, and by the way I noticed we shouldn't be calling r.Header.Add before checking for the error on creating the request!):

```go
req, _ := httputil.DumpRequestOut(r, true)                                                               
log.Debugf("Full analytics request: %v", string(req))
```

I think somehow the request from the UI is hitting the backend first and corrupting things.

**Comment by atavism** [comment]
@myleshorton Are you seeing it with the latest changes?

File: src/github.com/getlantern/analytics/analytics.go [hunk 1]
```diff
@@ -137,14 +138,15 @@ func SendRequest(payload *Payload) (status bool, err error) {
...
-	r.Header.Add("Content-Length", strconv.Itoa(len(args)))
 
 	if err != nil {
 		log.Errorf("Error constructing GA request: %s", err)
 		return false, err
 	}
 
+	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	r.Header.Add("Content-Length", strconv.Itoa(len(args)))
+
```
Comment (atavism) [inline src/github.com/getlantern/analytics/analytics.go]: Adding headers here only after we successfully construct the new request.

**Comment by myleshorton** [comment]
Yup working great @atavism thanks!

**Comment by myleshorton** [comment]
LGTM merging.

**Pull request merged**